ckanext/datastore/writer.py

```diff
@@ -11,6 +12,14 @@
 
 BOM = "\N{bom}"
 
+# (canada fork only): conform to XML stadards for element names
+# TODO: upstream contrib
+xml_element_name_rules = [
+    (re.compile(r'^([0-9]+xml|xml[0-9]+|xml|^[0-9]+)', re.I), ''),
```

Comment:
shorter and catches digits + "xml" + digits case
```suggestion
    (re.compile(r'^([0-9]*xml[0-9]*|[0-9]+)', re.I), ''),
```